test/tests/smoke_test.rs

```diff
@@ -12,18 +20,133 @@ fn assert_output_contains(haystack: &str, needle: &str) {
     );
 }
 
+#[test]
+fn retrieve_csr_test() {
+    const GENERATE_IDEVID_CSR: u32 = 1;
+    let rom = caliptra_builder::build_firmware_rom(&ROM_WITH_UART).unwrap();
+    let mut hw = caliptra_hw_model::new(BootParams {
+        init_params: InitParams {
+            rom: &rom,
+            security_state: *SecurityState::default().set_debug_locked(true),
+            ..Default::default()
+        },
+        initial_dbg_manuf_service_reg: GENERATE_IDEVID_CSR,
+        ..Default::default()
+    })
+    .unwrap();
+
+    let mut txn = hw.wait_for_mailbox_receive().unwrap();
+    let csr_der = mem::take(&mut txn.req.data);
+    txn.respond_success();
+
+    let csr = openssl::x509::X509Req::from_der(&csr_der).unwrap();
+    let csr_txt = String::from_utf8(csr.to_text().unwrap()).unwrap();
+
+    // To update the CSR testdata:
+    // std::fs::write("tests/smoke_testdata/idevid_csr.txt", &csr_txt).unwrap();
+    // std::fs::write("tests/smoke_testdata/idevid_csr.der", &csr_der).unwrap();
+
+    println!("csr: {}", csr_txt);
+
+    assert_eq!(
```

Comment:
Would it make sense to put some helpful panic message here, like "If your change is expected to modify generated certs or public keys, please re-generate the testdata (see comment in this file)"?

Reply:
Yes. Will do this in a followup PR.